indexing-service/src/main/java/io/druid/indexing/overlord/TaskStorage.java

```diff
@@ -53,6 +53,13 @@
   public void addLock(String taskid, TaskLock taskLock);
 
   /**
+   * Upgrades the lock state in the storage facility to non-preemptive lock
```

Comment:
the contract of this method is that if it *doesn't* throw an exception, everything is ok, right?